include/rear-panel/core/messages.hpp

```diff
@@ -339,29 +339,158 @@ struct DoorSwitchStateInfo
     auto operator==(const DoorSwitchStateInfo& other) const -> bool = default;
 };
 
+// NOLINTNEXTLINE(cppcoreguidelines-pro-type-member-init)
+struct AddLightActionRequest
+    : BinaryFormatMessage<rearpanel::ids::BinaryMessageId::add_light_action> {
+    static constexpr size_t LENGTH =
+        sizeof(uint16_t) + sizeof(rearpanel::ids::LightTransitionType) +
+        (4 * sizeof(uint8_t));
+    uint16_t length;
+    uint16_t transition_time_ms;
+    rearpanel::ids::LightTransitionType transition;
+    uint8_t red;
+    uint8_t green;
+    uint8_t blue;
+    uint8_t white;
+
+    template <bit_utils::ByteIterator Input, typename Limit>
+    static auto parse(Input body, Limit limit)
+        -> std::variant<std::monostate, AddLightActionRequest> {
+        uint16_t type = 0;
+        uint8_t transition_buf = 0;
+        auto ret = AddLightActionRequest{
+            .length = 0,
+            .transition_time_ms = 0,
+            .transition = rearpanel::ids::LightTransitionType::linear,
+            .red = 0,
+            .green = 0,
+            .blue = 0,
+            .white = 0};
```

Comment:
What do these values for each color represent? It's not the PWM that corresponds with each color, right?

Reply:
The Python definition of the message makes this more explicit, but these are uint8_t scaled from 0 to 255 to match standard RGB color definitions, which is how our design docs specify the colors to use. The max PWM is also scaled to 255 so we just keep the scale the same throughout the code.

Reply:
Talked offline 👍